token.go

```diff
@@ -58,15 +58,21 @@ func b64decode(data string) []byte {
 // Supports masked tokens. realToken comes from Token(r) and
 // sentToken is token sent unusual way.
 func VerifyToken(realToken, sentToken string) bool {
-	r := b64decode(realToken)
+	r, err := base64.StdEncoding.DecodeString(realToken)
+	if err != nil {
+		return false
+	}
 	if len(r) == 2*tokenLength {
 		r = unmaskToken(r)
 	}
-	s := b64decode(sentToken)
+	s, err := base64.StdEncoding.DecodeString(sentToken)
+	if err != nil {
+		return false
+	}
 	if len(s) == 2*tokenLength {
 		s = unmaskToken(s)
 	}
-	return subtle.ConstantTimeCompare(r, s) == 1
+	return subtle.ConstantTimeCompare(r, s) == 1 && len(r) > 0 && len(s) > 0
```

Comment:
Would a more restrictive check make sense, i.e. `len(r) == tokenLength && len(s) == tokenLength`? I think we can also move the length checks before slice comparison. [ConstantTimeCompare](https://golang.org/src/crypto/subtle/constant_time.go?s=505:546#L2) is already non-constant if slices have differing lengths.

Reply:
Also, it would probably be best to just delegate to `verifyToken` here after unmasking `realToken` (verifyToken expects one masked, the other unmasked :spaghetti: ). It already has such checks and there are less opportunities for footguns having one implementation rather than two.

Reply:
> Would a more restrictive check make sense, i.e. len(r) == tokenLength && len(s) == tokenLength? I think we can also move the length checks before slice comparison. ConstantTimeCompare is already non-constant if slices have differing lengths.

Will do! I'm busy with workshops until Friday so I won't be able to fix this before but am happy to do so on Friday :)

> Also, it would probably be best to just delegate to verifyToken here after unmasking realToken (verifyToken expects one masked, the other unmasked 🍝 ). It already has such checks and there are less opportunities for footguns having one implementation rather than two.

I'm not a 100% sure if I caught that but I have to say that I also don't 100% understand how masking/unmasking works and what they do. If you point me to the right areas I'd be happy to give it my best try but if you already have a concrete idea on how to do it feel free to go ahead :)